### PR TITLE
Add chopper timestamps to Workspace Log

### PR DESCRIPTION
**Pull request by LamarMoore**
**Description of work.**

The new [chopper schema](https://github.com/ess-dmsc/streaming-data-types/blob/master/schemas/tdct_timestamps.fbs) developed at the ESS has been included into the `KafkaEventStreamDecoder` so that all chopper timestamps can be added to the workspace log during streaming. This should allow for on-the-fly WFM mode date reduction. The chopper timestamps are stored as arrays of epoc based ticks (i.e `ArrayProperty<uint64_t>`). 

**To test:**

- Thorough code review.

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/LiveData/inc/MantidLiveData/Kafka/IKafkaStreamDecoder.tcc [hunk 1]
```diff
@@ -117,5 +124,50 @@ void IKafkaStreamDecoder::loadInstrument(const std::string &name,
                         "instrument geometry. \n";
   }
 }
+
+/**
+ * Add chopper timestamps to the mutable run info of all workspaces used to
+ * buffer data from the kafka stream.
+ * @param qorkspaces buffer workspaces storing kafka data.
```
Comment (matthew-d-jones) [inline Framework/LiveData/inc/MantidLiveData/Kafka/IKafkaStreamDecoder.tcc]: typo

**Review by matthew-d-jones — COMMENTED** [review]
I'm currently building the branch to test against the ISIS production cluster.
Code looks good to me, given the urgency of having this feature I think we should go ahead and merge if it doesn't break existing functionality. However it would be good to have a ticket to add chopper timestamp data to the tests.

**Comment by LamarMoore** [comment]
https://github.com/mantidproject/mantid/issues/26706

**Review by matthew-d-jones — CHANGES_REQUESTED** [review]
To avoid breaking the listener for users at ISIS the algorithm should not terminate if the `_choppers` topic does not exist.

**Review by matthew-d-jones — APPROVED** [review]
Great, I'm happy with the code and have checked changes do not interfere with existing use at ISIS. :+1: 
Tested against NeXus-Streamer.

**Pull request merged**